core/meta/src/TGenericClassInfo.cxx

```diff
@@ -27,31 +27,41 @@
 namespace ROOT {
 namespace Internal {
    std::string TTypeNameExtractionBase::GetImpl(const char* derived_funcname) {
```

Comment:
Note, this function still doesn't work on Windows, and now my changes make ROOT error out if type deduction fails, so Windows builds are going to fail. In the long term, I would prefer to get and use the mangled name as is, without parsing compiler output. Parsing like done here is quite unreliable, as this function itself proves. @bellenot , could you please let us know what Visual Studio prints so we can fix it?

Reply:
Nevermind, it already failed with
```
Fatal in <TClassTable::Add()>: Failed to deduce type for 'class ROOT::Internal::TCheckHashRecurveRemoveConsistency'
```
So I will add code to gobble up `"class "` in the beggining of the type string and see if that works.

Reply:
Hi @bellenot. Thanks. ROOT would just add the types that failed with an empty string as the key before. With ICC that started to generate lots of warnings and test failures, so I turned it into a fatal error.

Reply:
> In the long term, I would prefer to get and use the mangled name as is, without parsing compiler output. 

Agreed.  It is 'just' a matter of using TClassEdit::DemangleTypeIdName (on the right type :) ), so this might be straightforward enough for this PR.

Reply:
I will try, but why are we doing it the way it is now if it was just a matter of using `TClassEdit::DemangleTypeIdName`? I do not know what is needed as the key, so I did not try to change it. That is, I don't know if changing it from mangled to demandled has any effect, if it just needs to be something unique, etc. I need you and @Axel-Naumann to help me here.

Reply:
>  I do not know what is needed as the key

We need the demangled fully qualified name of the class (T) to be returned by TTypeNameExtraction<T>::Get().  A priori this could be accomplish by passing the typeid(T) to GetImp and have GetImp call TClassEdit::DemangleTypeIdName.

> That is, I don't know if changing it from mangled to demandled has any effect

The name currently returned is demangled.

> why are we doing it the way it is now

That seems to have been lost in time ... It is plausible that at the time the author concluded that TClassEdit could not be used from Rtypes.h (possibly due to the way the files were organized at the time).   We probably should still not #include TClassEdit.h in Rtypes.h but we can have TGenericClassInfo.cxx do it safely.

Reply:
What about this? Is it correct behavior?
```
   ------------------------------------------------------------
  | Welcome to ROOT 6.11/02                http://root.cern.ch |
  |                               (c) 1995-2017, The ROOT Team |
  | Built for linuxx8664gcc                                    |
  | From tag v6-11-02, 6 October 2017                          |
  | Try '.help', '.demo', '.license', '.credits', '.quit'/'.q' |
   ------------------------------------------------------------

root [0] int status;
root [1] cout << TClassEdit::DemangleTypeIdName(typeid(Double32_t), status) << endl;
double
```

Reply:
Answer = yes?
```
root [5] cout << ROOT::Internal::TTypeNameExtraction<Double32_t>::Get() << endl;
double
```

Reply:
> What about this? Is it correct behavior?

Unfortunately it is not ... However you point out it is no worse that the current situation.

Either way this means that ClassDefInline is currently *not* supported for class template whose argument might be Double32_t